web-services-app/src/main/java/edu/unc/lib/boxc/web/services/rest/SingleUseKeyController.java

```diff
@@ -86,17 +83,15 @@ public ResponseEntity<InputStream> download(@PathVariable("key") String accessKe
                 var principals = getAgentPrincipals().getPrincipals();
 
                 singleUseKeyService.invalidate(accessKey);
-                fedoraContentService.streamData(pid, datastream, principals, true, response);
+                fedoraContentService.streamData(pid, datastream, true, response);
                 log.info("Single use link used. Access Key: {}, UUID: {}", accessKey, id);
                 analyticsTracker.trackEvent(request, "download", pid, principals);
             } else {
                 throw new NotFoundException("Single use key is not valid: " + accessKey);
             }
-        } catch (Exception e) {
+        } catch (IOException e) {
             log.error("Download single use link did not work:", e);
-            throw new NotFoundException("Single use key is not valid: " + accessKey);
+            throw new RepositoryException("Single use key is not valid: " + accessKey);
```

Comment:
The message probably isn't accurate anymore, can make it a generic "Failed to download file using single use key" or something like that.